### ISSUE TITLE
Proposal: <podcast:alternateDescription> tag

### ISSUE THREAD

**Issue by Atreyu-94** [Podcastindex-org/podcast-namespace#372]
My proposal is fundamentally based on the internationalization of podcasts.

Sometimes it would be useful to have a version of the description in one or more languages. I more or less followed the idea of  ​​`<podcast:transcript> `.

The idea is the following:

### Parent

  `<item>`

### Count

  Multiple

### Attributes
- **url: (required)** The URL of the file containing the description. 
- **type: (required)** Mime type of the file such as `text/plain`, `text/html`. _I don't know exactly if there is a standard for formatting the description, at the moment HTML and Markdown come to mind_.
- **language: (required)** An [IETF language tag (BCP 47)](https://en.wikipedia.org/wiki/BCP_47) code identifying the language of this description.
- **title: (optional)** A translation of the podcast title into the language specified in the `lang` attribute. _It would be important to determine the length of the string in this case_.

### Examples
```html
<podcast:alternateDescription url="https://example.com/episode1/es/description.html" type="text/html" language="es" title="Este es el episodio 1 en español" />
<podcast:alternateDescription url="https://example.com/episode1/fr/description.html" type="text/html" language="fr" title="C'est l'épisode 1 en français" />
```
The original description should be considered with the same language that is specified by the RSS <language> element.

**Comment by jamescridland** [comment]
This is an interesting proposal. Permit me two pieces of feedback...

## type

The `<description>` de-facto standard is text with optional, limited support for simple HTML - [this research](https://podnews.net/article/html-episode-notes-in-podcast-rss) suggests that really only `<p>` and `<a href=` are supported. I'd suggest that we remove this value, and suggest that it be rendered in the same way as the `<description>` field.

The maximum length of the description is, in practice, 4,000 bytes. That's not written into any specification either; but for a specification, that might be a recommendation.

## language

Going very deep into the weeds here, the [RSS 2.0 specification for language tags](https://www.rssboard.org/rss-language-codes) is rather simpler than the BCP 47 or RFC5646 recommendations for HTML.

I would recommend aligning with the RSS 2.0 specification here, which is essentially [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) with a country code where required. That is significantly simpler for directories to parse, and aligns with the intentions of the RSS specification. I hate to wonder quite what a typical parser would do with `language="gsw-u-sd-chzh"` which is technically correct under RFC5646, but for the purposes of this specification would be rather more clearly identifiable as `language="de-CH"` (it's the particular dialect of German that they speak in Zürich, Switzerland).

Additionally, most usage on the web would expect `lang=` and not `language=` as a value within a tag. I've a slight preference for `lang=` since it's shorter and I'm lazy.

Finally, could I suggest that your French example uses `language="fr-CA"` to make the variations of language codes obvious for dim coders like me (i.e. the language field should be five characters long in your database, not just two).

**Comment by theDanielJLewis** [comment]
I’m sorry, I just don’t see the need for this.

Why should the description be in a language that’s _different_ from the podcast content? And if there’s actually a legitimate reason for that, then why only the description and not the rest of the RSS text, like show/episode titles, show/episode descriptions, chapters, and more?

I really think the better solution to this would be simply to release a different RSS feed with _all_ the text in the target language.

But that still goes back to the question of _why_ publish content in a language different from all the contextual text?

If it was a language-learning podcast, I still think it would be ideal for the majority of the text to be in the audience’s language, with only small quoted sections being in the other language. For example, an episode could be “What does ‘cinco’ mean?” Or a description could say, “Learn to be comfortable with common questions and answers useful in traveling, like ‘…,’ ‘…,’ and ‘….’”

**Comment by ryan-lingopods** [comment]
> Why should the description be in a language that’s _different_ from the podcast content?

I agree it would help to look at the actual use cases before deciding on what the feature should look like. i.e. Why do we actually want to have the description in a language that's different from the podcast audio?

Surveying multilingual podcasts, I can see various categories of podcasts that may mix languages (Also see the related issue  #367 - multilingual transcripts, which discusses the `xml:lang` attribute).

We have podcasts that are naturally multilingual, in regions where the audience commonly understands and switches between multiple languages on a daily basis. Interviews on each episode may be in the interviewee's native language, or sometimes switch mid conversation.

We also have language learning podcasts where a host might teach a target language using a source language, or sometimes teach 100% in the target language with the description in the source language.

That last one could be a valid use case for having multiple translations for the description. An example would be where the audio is 100% in simplified German, but intended to be understood by learners of German regardless of their source language, and in this case the description could theoretically be translated into multiple languages for multiple target audiences.

In this use case, I see benefits to @theDanielJLewis 's proposal of having a separate RSS feed per language, because if you are going to have 10 translations of the feed content for different target audiences, the size of your RSS feed will go up by 10x. Separating each language into a separate file means that the client can download just the translation that it needs depending on the user's locale.

On the other hand, consider podcasts that are naturally bilingual where the podcast hosts switch back and forth between two languages. In that case, having a separate RSS feed per language isn't the right solution, because the target audience is intended to understand both languages, and we would want to keep both languages in the description. Having something like @Atreyu-94 's proposal might then make more sense as at least part of the solution.

Regarding `language=` vs `lang=`, this was also discussed in #367 . There is prior art, since atom feeds use the existing `xml:lang` attribute to define the language content of any element, and that namespace can also be applied to RSS since RSS is an application of XML. Spec:

https://www.w3.org/TR/REC-xml/#sec-lang-tag

A minor point, but my preference is also to name the proposed tag `<podcast:description>` rather than `podcast:alternateDescription`>. If we were were designing the RSS spec from scratch, that would have been the way I would design it, since that single tag would satisfy all use cases. It would also work in the case where a feed has only a single description. It would also make sense where each translation is split into a separate file so that effectively each RSS feed has a single description (with a different `xml:lang` attribute).

**Comment by theDanielJLewis** [comment]
How about approaching this differently? Like a `<podcast:translation>` or similar tag. Its purpose would be to indicate the overrides for any included tags for the associated channel or item. The translation tag itself would indicate the language and provide a URL for the alternate translation. Then, the only required tags would be the `<guid>` for the channel and items. Any additional tags would take precedence over the master RSS's content.

For example, the translated feed might look as simple as this for episodes:

```xml
<item>
  <guid>123455</guid>
  <title>Español</title>
</item>
```

When an item looks like that, it would replace _only_ the title of the item that shares the GUID; all other text and content would remain the same. This could even allow for overwriting things like chapters and transcripts.

This _does_, however, present a potential conflict with the possibility of an enclosure tag in the translated feed. Maybe that would be better than alternateEnclosure, or maybe it would be worse. But I think we could say simply, "Do not include enclosure tags in the translated feed."
